orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatModel.java

```diff
@@ -0,0 +1,71 @@
+package com.sap.ai.sdk.orchestration.spring;
+
+import com.sap.ai.sdk.orchestration.AssistantMessage;
+import com.sap.ai.sdk.orchestration.OrchestrationClient;
+import com.sap.ai.sdk.orchestration.OrchestrationPrompt;
+import com.sap.ai.sdk.orchestration.SystemMessage;
+import com.sap.ai.sdk.orchestration.UserMessage;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.ai.chat.messages.Message;
+import org.springframework.ai.chat.model.ChatModel;
+import org.springframework.ai.chat.model.ChatResponse;
+import org.springframework.ai.chat.prompt.Prompt;
+
+/** Spring AI integration for the orchestration service. */
+@Slf4j
+@RequiredArgsConstructor
+public class OrchestrationChatModel implements ChatModel {
+  @Nonnull private OrchestrationClient client;
+
+  /** Default constructor. */
+  public OrchestrationChatModel() {
+    this.client = new OrchestrationClient();
+  }
+
+  @Nonnull
+  @Override
+  public ChatResponse call(@Nonnull final Prompt prompt) {
+
+    if (prompt.getOptions() != null
+        && prompt.getOptions() instanceof OrchestrationChatOptions options) {
+
+      val orchestrationPrompt = toOrchestrationPrompt(prompt);
+      val response = client.chatCompletion(orchestrationPrompt, options.getConfig());
+      return OrchestrationChatResponse.fromOrchestrationResponse(response.getOriginalResponse());
+    }
+    throw new IllegalArgumentException(
+        "Please add OrchestrationChatOptions to the Prompt: new Prompt(\"message\", new OrchestrationChatOptions(config))");
```

Comment:
(Question)

So we're planning to throw an exception, when the provided `prompt` argument does not contain an `OrchestrationChatOptions` object. This is implicit behavior. Users may not catch that limitation on design-time.

Would it be possible to make it explicit, e.g. by enforcing a constructor argument for `OrchestrationChatOptions` on `OrchestrationChatModel`? Thus passing it on for every `call`?

Reply:
call is part if the interface, cannot be modified.
If we do it in the constructor then there are 2 different places to add a config. We have to merge them which is implicit behaviour.

This is not solveable

Reply:
If not constructor - offering an optional `setDefaultChatOptions(..)` is probably not helping either, right? Because then you would need to thin about merging options ._.